### DISCUSSION TITLE
iOS: GLBs in Cache affects Quick Lock Performance

### DISCUSSION THREAD

**Discussion by phreak** [google/model-viewer#4853] [General]
Hi all,

We recently encountered issues with Quick Look AR on iPhones, where the AR experience wouldn't start sometimes, showing an endless loading wheel instead. If users closed Quick Look during this process, it would render the AR button in Model Viewer (MV) non-functional. The only solution was to force-close Safari by swiping it up in the iOS app switcher. Simply reloading the browser window didn't restore the AR button's functionality.

Through various tests and optimization scenarios, we discovered that previously loaded GLB files (in Model Viewer) were somehow interfering with the iPhone's cache. We confirmed this by testing against a scenario where we removed the GLB files from the folder, (boldly accepting 404s to test). Interestingly, in this case, Quick Look worked fine.

Anyone came across similar behaviour, found a fix for this.

- iPhone 14 Pro
- iOS: 176
- GLBs validate proper with 5MB, USDZs 15 MB.

**Comment by elalish** [comment]
No, that's a new one. Do you have a URL where we can repro? @AdaRoseCannon is this a WebKit or QuickLook bug?
